### PR TITLE
Add an option to get the output in plain, json or print_r syntax

### PR DESCRIPTION
**Pull request by nickvergessen**
Follow up for #15448 

@DeepDiver1975 @MorrisJobke @jnweiger 
### ./occ check

```
nickv@nickv-lt-mint $ ./occ check
 - Setting locale to en_US.UTF-8/fr_FR.UTF-8/es_ES.UTF-8/de_DE.UTF-8/ru_RU.UTF-8/pt_BR.UTF-8/it_IT.UTF-8/ja_JP.UTF-8/zh_CN.UTF-8 failed

nickv@nickv-lt-mint $ ./occ check --output=print
Array
(
    [0] => Setting locale to en_US.UTF-8/fr_FR.UTF-8/es_ES.UTF-8/de_DE.UTF-8/ru_RU.UTF-8/pt_BR.UTF-8/it_IT.UTF-8/ja_JP.UTF-8/zh_CN.UTF-8 failed
)

nickv@nickv-lt-mint $ ./occ check --output=json
["Setting locale to en_US.UTF-8\/fr_FR.UTF-8\/es_ES.UTF-8\/de_DE.UTF-8\/ru_RU.UTF-8\/pt_BR.UTF-8\/it_IT.UTF-8\/ja_JP.UTF-8\/zh_CN.UTF-8 failed"]
```
### ./occ status

```
nickv@nickv-lt-mint $ ./occ status
 - installed: true
 - version: 8.1.0.1
 - versionstring: 8.1 pre alpha
 - edition: Enterprise
nickv@nickv-lt-mint $ ./occ status --output=print
Array
(
    [installed] => true
    [version] => 8.1.0.1
    [versionstring] => 8.1 pre alpha
    [edition] => Enterprise
)

nickv@nickv-lt-mint $ ./occ status --output=json
{"installed":"true","version":"8.1.0.1","versionstring":"8.1 pre alpha","edition":"Enterprise"}
```
### ./occ app:list

```
nickv@nickv-lt-mint ~/ownCloud/master/core {master*}:
$ ./occ app:list
Enabled:
 - files: 1.1.9
 - files_external: 0.2.3
 - files_sharing: 0.6.1
 - files_texteditor: 0.4
 - files_trashbin: 0.6.2
 - files_versions: 1.0.5
 - provisioning_api: 0.2
Disabled:
 - activity: 1.2.3
 - encryption: 
 - encryption_dummy: 
nickv@nickv-lt-mint ~/ownCloud/master/core {master*}:
$ ./occ app:list --output=print
Array
(
    [enabled] => Array
        (
            [files] => 1.1.9
            [files_external] => 0.2.3
            [files_sharing] => 0.6.1
            [files_texteditor] => 0.4
            [files_trashbin] => 0.6.2
            [files_versions] => 1.0.5
            [provisioning_api] => 0.2
        )

    [disabled] => Array
        (
            [activity] => 1.2.3
            [encryption] => 
            [encryption_dummy] => 
        )

)
nickv@nickv-lt-mint ~/ownCloud/master/core {master*}:
$ ./occ app:list --output=json
{"enabled":{"files":"1.1.9","files_external":"0.2.3","files_sharing":"0.6.1","files_texteditor":"0.4","files_trashbin":"0.6.2","files_versions":"1.0.5","provisioning_api":"0.2"},"disabled":{"activity":"1.2.3","encryption":"","encryption_dummy":false}}

```

### CODE REVIEW

File: core/command/status.php [hunk 1]
```diff
@@ -41,6 +42,7 @@ protected function execute(InputInterface $input, OutputInterface $output) {
 			'versionstring' => \OC_Util::getVersionString(),
 			'edition' => \OC_Util::getEditionString(),
 		);
-		print_r($values);
+
+		$this->writeArrayInOutputFormat($input, $output, $values);
```
Comment (nickvergessen) [inline core/command/status.php]: @DeepDiver1975 I changed the default behaviour here, it now outputs a normal list, in order to get teh old print_r syntax you need to use `--output=print`. Is that okay?

```
nickv@nickv-lt-mint $ ./occ status
 - installed: true
 - version: 8.1.0.1
 - versionstring: 8.1 pre alpha
 - edition: Enterprise
nickv@nickv-lt-mint $ ./occ status --output=print
Array
(
    [installed] => true
    [version] => 8.1.0.1
    [versionstring] => 8.1 pre alpha
    [edition] => Enterprise
)

File: core/command/app/listapps.php [hunk 2]
```diff
@@ -51,13 +54,38 @@ protected function execute(InputInterface $input, OutputInterface $output) {
...
+			if (isset($versions[$app])) {
+				$apps['disabled'][$app] = $versions[$app];
+			} else {
+				$apps['disabled'][$app] = false;
+			}
+		}
+		$this->writeArrayInOutputFormat($input, $output, $apps);
+	}
+
+	protected function writeArrayInOutputFormat(InputInterface $input, OutputInterface $output, $items) {
```
Comment (DeepDiver1975) [inline core/command/app/listapps.php]: duplicate

File: core/command/status.php (hunk 1 above)
Comment (DeepDiver1975) [inline core/command/status.php]: sure - https://github.com/owncloud/core/commit/b9d863bc5f34ee0ad9da5ab6be0d072d17149d90 :wink:

File: core/command/base.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+	protected function writeArrayInOutputFormat(InputInterface $input, OutputInterface $output, $items) {
+		$outputFormat = $input->getOption('output');
+		switch ($outputFormat) {
+			case 'json':
+			case 'print':
+				if ($outputFormat === 'json') {
+					$output->writeln(json_encode($items));
+				} else {
+					print_r($items);
```
Comment (DeepDiver1975) [inline core/command/base.php]: We should only use $output - no echo, print* or what so ever

File: core/command/base.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+use Symfony\Component\Console\Output\OutputInterface;
+
+class Base extends Command {
+	protected function configure() {
+		$this
+			->addOption(
+				'output',
+				null,
+				InputOption::VALUE_OPTIONAL,
+				'Output format (plain, print or json, default is plain)',
```
Comment (nickvergessen) [inline core/command/base.php]: Could also add json_pretty aswell here

File: core/command/status.php (hunk 1 above)
Comment (nickvergessen) [inline core/command/status.php]: okay, then I will drop print in favor of json_pretty

**Comment by karlitschek** [comment]
:+1:

**Comment by MorrisJobke** [comment]
Nice and works :+1:

**Pull request merged**